Explorer/Assets/Scripts/ECS/StreamableLoading/Common/Systems/LoadSystemBase.cs

```diff
@@ -85,18 +91,25 @@ protected override void Update(float t)
 
         private void Execute(in Entity entity, ref TIntention intention)
         {
+            if (!intention.IsAllowed())
+                return;
+
             // Remove current source flag from the permitted sources
             // it indicates that the current source was used
             intention.RemoveCurrentSource();
 
             // Try load from cache first
             if (TryLoadFromCache(in entity, in intention))
+            {
+                concurrentLoadingBudgetProvider?.ReleaseBudget();
```

Comment:
let's forbid passing a null budget provider and introduce a `NULL-pattern` provider that always allows by default